pafy/channel.py

```diff
@@ -0,0 +1,232 @@
+import re
+from .pafy import call_gdata
+from .playlist import Playlist
+from .backend_shared import pyver
+
+
+def get_channel(channel_url, basic=False, gdata=False,
+                size=False, callback=None):
+    """Return a Channel object
+
+    The returned Pafy and Playlist objects are initialised using the arguments
+    to get_channel() in the manner documented for pafy.new()
+
+    """
+
+    return Channel.from_url(channel_url, basic, gdata, size, callback)
+
+
+class Channel(object):
+    def __init__(self, channel_url, basic, gdata, size, callback):
+
+        self._channel_url = channel_url
+        self._channel_id = None
+        self._title = None
+        self._description = None
+        self._logo = None
+        self._subscriberCount = None
+        self._uploads = None
+        self._basic = basic
+        self._gdata = gdata
+        self._size = size
+        self._callback = callback
+        self._playlists = None
+        self._subscriptions = None
+        self._have_basic = False
+
+    @classmethod
+    def from_dict(cls, ch, basic, gdata, size, callback):
+        t = cls(ch['id'], basic, gdata, size, callback)
+        t._channel_id = ch['id']
+        t._title = ch['title']
+        t._description = ch['description']
+        t._logo = ch['logo']
+        t._subscriberCount = ch['subscriberCount']
+        t._uploads = ch['uploads']
+        t._have_basic = True
+
+        return t
+
+    @classmethod
+    def from_url(cls, url, basic, gdata, size, callback):
+        t = cls(url, basic, gdata, size, callback)
+        t._fetch_basic()
+        return t
+
+    @property
+    def channel_id(self):
+        if not self._have_basic:
+            self._fetch_basic()
+        return self._channel_id
+
+    @property
+    def title(self):
+        if not self._have_basic:
+            self._fetch_basic()
+        return self._title
+
+    @property
+    def description(self):
+        if not self._have_basic:
+            self._fetch_basic()
+        return self._description
+
+    @property
+    def logo(self):
+        if not self._have_basic:
+            self._fetch_basic()
+        return self._logo
+
+    @property
+    def subscriberCount(self):
+        if not self._have_basic:
+            self._fetch_basic()
+        return self._subscriberCount
+
+    @property
+    def uploads(self):
+        if not self._uploads:
+            self._fetch_basic()
+        if type(self._uploads) != Playlist:
+            self._uploads = Playlist.from_url(self._uploads, self._basic,
+                                              self._gdata, self._size,
+                                              self._callback)
+
+        return self._uploads
+
+    @property
+    def playlists(self):
```

Comment:
I think it would be better to implement this as a generator, so iterating over it you don't need to retrieve multiple pages of results unless/until they are needed. See how `__iter__()` is implemented for `Playlist`.